tests/RobotFramework/tests/cumulocity/registration/registration_lifecycle.robot

```diff
@@ -30,6 +30,19 @@ Child device registration
     Cumulocity.Set Device    ${DEVICE_SN}
     Cumulocity.Should Be A Child Device Of Device    ${CHILD_SN}
 
+    # Deregister Child device
+    Execute Command    mosquitto_sub --remove-retained -W 3 -t "te/device/${CHILD_SN}/+/+/#"    exp_exit_code=27
+    
+    # Check if deregistration was successful
+    ${retained_message_count}=    Get Retained Message Count    te/device/${CHILD_SN}/+/+/#
```

Comment:
An explicit sleep is used in the deregister service case, so by the same logic it should be used here as well.

For example
```suggestion
    Sleep    1s    reason=Allowing components to process messages
    ${retained_message_count}=    Get Retained Message Count    te/device/${CHILD_SN}/+/+/#
```